extensions/blocks/eventbrite/edit.js

```diff
@@ -0,0 +1,393 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { __, _x } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import {
+	Placeholder,
+	SandBox,
+	Button,
+	IconButton,
+	Toolbar,
+	PanelBody,
+	Spinner,
+	ExternalLink,
+} from '@wordpress/components';
+import { BlockControls, BlockIcon, BlockPreview } from '@wordpress/block-editor';
+import { withDispatch } from '@wordpress/data';
+import { InspectorControls } from '@wordpress/editor';
+import apiFetch from '@wordpress/api-fetch';
+import { ENTER, SPACE } from '@wordpress/keycodes';
+import { getBlockFromExample } from '@wordpress/blocks';
+/**
+ * Internal dependencies
+ */
+import { createWidgetId, convertToLink, eventIdFromUrl } from './utils';
+import { CUSTOM_URL_REGEX, icon, URL_REGEX } from '.';
+import ModalButtonPreview from './modal-button-preview';
+import EventbriteInPageExample from './eventbrite-in-page-example.png';
+import './editor.scss';
+
+const MODAL_BUTTON_STYLES = [
+	{ name: 'fill', label: __( 'Fill' ), isDefault: true },
+	{ name: 'outline', label: __( 'Outline' ) },
+];
+
+class EventbriteEdit extends Component {
```

Comment:
Is there a reason for using a class component over a functional one and hooks? Probably just a matter of taste, but Gutenberg seems to be moving towards all functional components.

Reply:
I haven't worked on this since the start, and while I agree with you, I'd argue that we're too close to shipping to justify such large refactor. 🙂

Reply:
I agree, maybe just something worth keeping in mind for future blocks :)

Reply:
This started out as a direct copy and paste from the Pinterest block, that's the only reason--will definitely keep functional components in mind for the next block.